src/java/fr/paris/lutece/plugins/forms/util/FormResponseUtils.java

```diff
@@ -0,0 +1,385 @@
+    
+    private FormResponseUtils( )
+    {
+    }
+    
+    /**
+     * @return The next Step
+     */
+    public static Step getNextStep( Step currentStep, List<String> errorList, FormResponseManager formResponseManager )
+    {
+        List<Transition> listTransition = TransitionHome.getTransitionsListFromStep( currentStep.getId( ) );
+
+        for ( Transition transition : listTransition )
+        {
+            List<Control> listTransitionControl = ControlHome.getControlByControlTargetAndType( transition.getId( ), ControlType.TRANSITION );
+            boolean controlsValidated = true;
+
+            if ( listTransitionControl.isEmpty( ) )
+            {
+                return StepHome.findByPrimaryKey( transition.getNextStep( ) );
+            }
+
+            for ( Control transitionControl : listTransitionControl )
+            {
+                Question targetQuestion = QuestionHome.findByPrimaryKey( transitionControl.getListIdQuestion( ).iterator( ).next( ) );
+                Step stepTarget = StepHome.findByPrimaryKey( targetQuestion.getIdStep( ) );
+                List<FormQuestionResponse> listQuestionResponse = formResponseManager.findResponsesFor( stepTarget ).stream( )
+                        .filter( q -> transitionControl.getListIdQuestion( ).stream( ).anyMatch( t -> t.equals( q.getQuestion( ).getId( ) ) ) )
+                        .collect( Collectors.toList( ) );
+
+                IValidator validator = EntryServiceManager.getInstance( ).getValidator( transitionControl.getValidatorName( ) );
+                if ( validator != null && !validator.validate( listQuestionResponse, transitionControl ) )
+                {
+                    controlsValidated = false;
+                    errorList.add( transitionControl.getErrorMessage( ) );
+                    break;
+                }
+            }
+
+            if ( controlsValidated )
+            {
+                return StepHome.findByPrimaryKey( transition.getNextStep( ) );
+            }
+
+        }
+
+        return null;
+    }
+    
+    public static FormResponseManager initFormResponseManager( HttpServletRequest request, Form form, FormResponseManager formResponseManager, boolean frontOffice )
+    {
+    	if ( formResponseManager == null )
+        {
+    		if ( frontOffice )
+        	{
+        		LuteceUser user = SecurityService.getInstance( ).getRegisteredUser( request );
+
+                if ( user != null )
+                {
+                	formResponseManager = _formService.createFormResponseManagerFromBackUp( form, user.getName( ) );
+                }
+                else
+                {
+                	formResponseManager = new FormResponseManager( form );
+                }
+        	}
+        	else
+        	{
+        		User admin = AdminUserService.getAdminUser( request );
+        		
+        		formResponseManager = _formService.createFormResponseManagerFromBackUp( form, admin.getEmail( ) );

```

Comment:
the service you are calling takes the front user's guid as a parameter, why are you passing the admin's email?